test_runner/fixtures/types.py

```diff
@@ -0,0 +1,109 @@
+import random
+from functools import total_ordering
+from typing import Union
+
+
+@total_ordering
+class Lsn:
+    """
+    Datatype for an LSN. Internally it is a 64-bit integer, but the string
+    representation is like "1/123abcd". See also pg_lsn datatype in Postgres
+    """
+
+    def __init__(self, x: Union[int, str]):
+        if isinstance(x, int):
+            self.lsn_int = x
+        else:
+            """Convert lsn from hex notation to int."""
+            l, r = x.split("/")
+            self.lsn_int = (int(l, 16) << 32) + int(r, 16)
+            # FIXME: error if it doesn't look like a valid LSN
+
+    def __repr__(self):
+        """Convert lsn from int to standard hex notation."""
+        return "{:X}/{:X}".format(self.lsn_int >> 32, self.lsn_int & 0xFFFFFFFF)
+
+    def __int__(self):
+        return self.lsn_int
+
+    def __lt__(self, other: "Lsn") -> bool:
+        return self.lsn_int < other.lsn_int
+
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, Lsn):
+            return NotImplemented
+        return self.lsn_int == other.lsn_int
+
+    # Returns the difference between two Lsns, in bytes
+    def __sub__(self, other: "Lsn") -> int:
+        return self.lsn_int - other.lsn_int
+
+    def __hash__(self):
+        return hash(self.lsn_int)
+
+
+@total_ordering
+class ZTenantId:
+    """
+    Datatype for a Neon tenant ID. Internally it's a 16-byte array, and
+    the string representation is in hex. This corresponds to the ZTenantId
+    type in the Rust code.
+    """
+
+    def __init__(self, x: str):
+        self.id = bytearray.fromhex(x)
+        assert len(self.id) == 16
+
+    def __repr__(self):
+        return self.id.hex()
```

Comment:
Nit: It's common practice for `__repr__` to contain all information necessary to reconstruct the object. Some error messages rely on that. In this case we get hex digits, but we don't know if it's a tenant id, timeline id, or maybe just a string.

Example
```
>>> uuid4().__str__()
'5649d3d5-e760-4dd4-8ddf-45a5e906e185'
>>> uuid4().__repr__()
"UUID('b3610abf-8883-48a7-82e4-00165a54bbdb')"
```

We can do something similar by implementing `__str__` in additino to `__repr__`.

Reply:
done